config/check_lfs_support.cpp: No such file or directory

11.1.2 fails to build for me:

```
CMake Error at CMakeLists.txt:55 (file):
   file failed to open for reading (No such file or directory):
 
     /home/abuild/rpmbuild/BUILD/nowide_v11.1.2/config/check_lfs_support.cpp
```

--- a/.appveyor.yml
+++ b/.appveyor.yml
@@ -102,7 +102,7 @@ environment:
       APPVEYOR_BUILD_WORKER_IMAGE: Visual Studio 2019
       GENERATOR: Visual Studio 16 2019
       configuration: Debug
-      BOOST_ROOT: C:\Libraries\boost_1_71_0
+      BOOST_ROOT: C:\Libraries\boost_1_73_0
       COVERAGE: true
       # Work around for https://community.codecov.io/t/bash-script-does-not-detect-appveyor/1094
       APPVEYOR: true
@@ -137,6 +137,7 @@ for:
       only: [CMAKE: true]
     install: true
     before_build:
+      - dir C:\Libraries
       - if exist build\ (rmdir /S /Q build)
       - set "INSTALL_DIR=%APPVEYOR_BUILD_FOLDER%\installed"
       - if exist %INSTALL_DIR%\ (rmdir /S /Q %INSTALL_DIR%)

--- a/.azure-pipelines.yml
+++ b/.azure-pipelines.yml
@@ -45,58 +45,89 @@ stages:
   jobs:
 
   - job: 'Linux'
-    pool:
-      vmImage: 'ubuntu-16.04'
     strategy:
       matrix:
+        GCC_10:
+          B2_TOOLSET: gcc
+          B2_CXXSTD: 14,17,20
+          CXX: g++-10
+          PACKAGES: g++-10
+          VM_IMAGE: ubuntu-20.04
+        GCC_9:
+          B2_TOOLSET: gcc
+          B2_CXXSTD: 14,17,2a
+          CXX: g++-9
+          PACKAGES: g++-9
+          VM_IMAGE: ubuntu-20.04
         GCC_8:
           B2_TOOLSET: gcc
           B2_CXXSTD: 14,17,2a
           CXX: g++-8
           PACKAGES: g++-8
+          VM_IMAGE: ubuntu-20.04
         GCC_7:
           B2_TOOLSET: gcc
           B2_CXXSTD: 11,14,17
           CXX: g++-7
           PACKAGES: g++-7
+          VM_IMAGE: ubuntu-18.04
         GCC_6:
           B2_TOOLSET: gcc
           B2_CXXSTD: 11,14
           CXX: g++-6
           PACKAGES: g++-6
+          VM_IMAGE: ubuntu-18.04
         GCC_5:
           B2_TOOLSET: gcc
           B2_CXXSTD: 11
           CXX: g++-5
           PACKAGES: g++-5
-        GCC_4_9:
-          B2_TOOLSET: gcc
-          B2_CXXSTD: 03,11
-          CXX: g++-4.9
-          PACKAGES: g++-4.9
+          VM_IMAGE: ubuntu-18.04
         GCC_4_8:
           B2_TOOLSET: gcc
           B2_CXXSTD: 03,11
           CXX: g++-4.8
           PACKAGES: g++-4.8
+          VM_IMAGE: ubuntu-18.04
         Clang_12:
           B2_TOOLSET: clang
           B2_CXXSTD: 14,17,20
           CXX: clang++-12
           PACKAGES: clang-12
-          LLVM_REPO: llvm-toolchain-xenial-12
+          VM_IMAGE: ubuntu-20.04
+        Clang_11:
+          B2_TOOLSET: clang
+          B2_CXXSTD: 14,17,20
+          CXX: clang++-11
+          PACKAGES: clang-11
+          VM_IMAGE: ubuntu-20.04
+        Clang_10:
+          B2_TOOLSET: clang
+          B2_CXXSTD: 14,17,20
+          CXX: clang++-10
+          PACKAGES: clang-10
+          VM_IMAGE: ubuntu-20.04
         Clang_9:
           B2_TOOLSET: clang
           B2_CXXSTD: 14,17,2a
           CXX: clang++-9
           PACKAGES: clang-9
-          LLVM_REPO: llvm-toolchain-xenial-9
+          VM_IMAGE: ubuntu-20.04
+        Clang_8:
+          B2_TOOLSET: clang
+          B2_CXXSTD: 14,17
+          CXX: clang++-8
+          PACKAGES: clang-8 libc6-dbg libc++-dev libstdc++-8-dev
+          LLVM_OS: bionic
+          LLVM_REPO: llvm-toolchain-bionic-8
+          VM_IMAGE: ubuntu-18.04
         Clang_7:
           B2_TOOLSET: clang
           B2_CXXSTD: 14,17
           CXX: clang++-7
           PACKAGES: clang-7
           LLVM_REPO: llvm-toolchain-xenial-7
+          VM_IMAGE: ubuntu-18.04
         Clang_6_libcxx:
           B2_TOOLSET: clang
           B2_CXXSTD: 03,11,14,17
@@ -104,34 +135,38 @@ stages:
           CXX: clang++-6.0
           PACKAGES: clang-6.0
           LLVM_REPO: llvm-toolchain-xenial-6.0
+          VM_IMAGE: ubuntu-18.04
         Clang_6:
           B2_TOOLSET: clang
           B2_CXXSTD: 14,17
           CXX: clang++-6.0
           PACKAGES: clang-6.0
           LLVM_REPO: llvm-toolchain-xenial-6.0
+          VM_IMAGE: ubuntu-18.04
         Clang_5:
           B2_TOOLSET: clang
           B2_CXXSTD: 11,14,17
           PACKAGES: clang-5.0
           CXX: clang++-5.0
           LLVM_REPO: llvm-toolchain-xenial-5.0
+          VM_IMAGE: ubuntu-18.04
         Clang_4:
           B2_TOOLSET: clang
           B2_CXXSTD: 11,14
           CXX: clang++-4.0
           PACKAGES: clang-4.0
           LLVM_REPO: llvm-toolchain-xenial-4.0
+          VM_IMAGE: ubuntu-18.04
         Clang_3_9:
           B2_TOOLSET: clang
           B2_CXXSTD: 03,11,14
           CXX: clang++-3.9
           PACKAGES: clang-3.9
-        Clang_3_5:
-          B2_TOOLSET: clang
-          B2_CXXSTD: 03,11
-          CXX: clang++-3.5
-          PACKAGES: clang-3.5
+          VM_IMAGE: ubuntu-18.04
+
+    pool:
+      vmImage: $(VM_IMAGE)
+
     steps:
     - bash: |
         set -e

--- a/.github/workflows/ci_tests.yml
+++ b/.github/workflows/ci_tests.yml
@@ -12,17 +12,17 @@ jobs:
   UnitTests:
     strategy:
       matrix:
-        os: [ubuntu-16.04, windows-latest]
+        os: [ubuntu-18.04, windows-latest]
         buildType: [Debug, Release]
         standalone: [Boost, Standalone]
         shared_lib: [ON, OFF]
         generator: ['Visual Studio 16 2019', 'MinGW Makefiles', 'Unix Makefiles']
         exclude:
-          - os: ubuntu-16.04
+          - os: ubuntu-18.04
             generator: MinGW Makefiles
-          - os: ubuntu-16.04
+          - os: ubuntu-18.04
             generator: Visual Studio 16 2019
-          - os: ubuntu-16.04
+          - os: ubuntu-18.04
             buildType: Debug
     runs-on: ${{matrix.os}}
     env:
@@ -103,7 +103,7 @@ jobs:
         ctest --output-on-failure -C ${{matrix.buildType}}
 
   BoostCMakeBuild:
-    runs-on: ubuntu-16.04
+    runs-on: ubuntu-18.04
     # Doesn't work on master yet
     if: "! endsWith(github.ref, '/master') && ! endsWith(github.base_ref, '/master')"
     steps:

--- a/.github/workflows/posix.yml
+++ b/.github/workflows/posix.yml
@@ -56,14 +56,27 @@ jobs:
       - name: Install packages
         if: matrix.install
         run: |
+          if [ -f "/etc/debian_version" ]; then
+              echo "DEBIAN_FRONTEND=noninteractive" >> $GITHUB_ENV
+              export DEBIAN_FRONTEND=noninteractive
+          fi
+          # Install required stuff when using containers
+          if [ -n "${{matrix.container}}" ]; then
+              if [ -f "/etc/debian_version" ]; then
+                  apt-get -o Acquire::Retries=$NET_RETRY_COUNT update
+                  apt-get -o Acquire::Retries=$NET_RETRY_COUNT install -y sudo software-properties-common build-essential g++ python
+                  add-apt-repository ppa:git-core/ppa # Need newer git
+              fi
+          fi
           if [[ "${{matrix.install}}" == "yes" ]]; then
             pkgs="${{matrix.compiler}}"
             pkgs="${pkgs/clang++-/clang-}"
           else
             pkgs="${{matrix.install}}"
           fi
+          sudo add-apt-repository ppa:ubuntu-toolchain-r/test
           sudo apt-get -o Acquire::Retries=$NET_RETRY_COUNT update
-          sudo apt-get -o Acquire::Retries=$NET_RETRY_COUNT install $pkgs
+          sudo apt-get -o Acquire::Retries=$NET_RETRY_COUNT install -y git $pkgs
 
       - name: Setup config vars
         run: |

--- a/.github/workflows/posix.yml
+++ b/.github/workflows/posix.yml
@@ -16,6 +16,7 @@ env:
 jobs:
   CI:
     strategy:
+      fail-fast: false
       matrix:
         include:
           - { compiler: g++-4.4,        cxxstd: '98,0x',                         os: ubuntu-16.04, install: yes }

--- a/.github/workflows/posix.yml
+++ b/.github/workflows/posix.yml
@@ -15,19 +15,23 @@ env:
 
 jobs:
   CI:
+    defaults:
+      run:
+        shell: bash
+
     strategy:
       fail-fast: false
       matrix:
         include:
-          - { compiler: g++-4.4,        cxxstd: '98,0x',                         os: ubuntu-16.04, install: yes }
-          - { compiler: g++-5,          cxxstd: '03,11,14',                      os: ubuntu-16.04, install: yes }
-          - { compiler: g++-6,          cxxstd: '03,11,14,17',                   os: ubuntu-16.04, install: yes }
+          - { compiler: g++-4.4,        cxxstd: '98,0x',                         os: ubuntu-20.04, container: 'ubuntu:16.04', install: yes }
+          - { compiler: g++-5,          cxxstd: '03,11,14',                      os: ubuntu-18.04, install: yes }
+          - { compiler: g++-6,          cxxstd: '03,11,14,17',                   os: ubuntu-18.04, install: yes }
           - { compiler: g++-7,          cxxstd: '03,11,14,17',                   os: ubuntu-20.04, install: yes }
           - { compiler: g++-8,          cxxstd: '03,11,14,17,2a',                os: ubuntu-20.04, install: yes }
           - { compiler: g++-9,          cxxstd: '03,11,14,17,2a',                os: ubuntu-20.04, install: yes }
           - { compiler: g++-10,         cxxstd: '03,11,14,17,2a', sanitize: yes, os: ubuntu-20.04, install: yes, linkflags: -fuse-ld=gold }
-          - { compiler: clang++-3.5,    cxxstd: '03,11',                         os: ubuntu-16.04, install: yes }
-          - { compiler: clang++-6.0,    cxxstd: '03,11',                         os: ubuntu-16.04, install: yes }
+          - { compiler: clang++-3.5,    cxxstd: '03,11',                         os: ubuntu-20.04, container: 'ubuntu:16.04', install: yes }
+          - { compiler: clang++-6.0,    cxxstd: '03,11',                         os: ubuntu-18.04, install: yes }
           - { compiler: clang++-7,      cxxstd: '03,11',                         os: ubuntu-18.04, install: yes }
           - { compiler: clang++-8,      cxxstd: '03,11,14',                      os: ubuntu-18.04, install: yes }
           - { compiler: clang++-9,      cxxstd: '03,11,14',                      os: ubuntu-18.04, install: yes }
@@ -38,6 +42,7 @@ jobs:
           - { compiler: g++-8,          cxxstd: '03,11',          coverage: yes, os: ubuntu-20.04, install: yes }
 
     runs-on: ${{matrix.os}}
+    container: ${{matrix.container}}
 
     steps:
       - uses: actions/checkout@v2

--- a/.github/workflows/posix.yml
+++ b/.github/workflows/posix.yml
@@ -9,6 +9,7 @@ on:
       - feature/**
 
 env:
+  NET_RETRY_COUNT: 5
   UBSAN_OPTIONS: print_stacktrace=1
   B2_VARIANT: debug,release
   B2_LINK: shared,static
@@ -61,7 +62,8 @@ jobs:
           else
             pkgs="${{matrix.install}}"
           fi
-          sudo apt install $pkgs
+          sudo apt-get -o Acquire::Retries=$NET_RETRY_COUNT update
+          sudo apt-get -o Acquire::Retries=$NET_RETRY_COUNT install $pkgs
 
       - name: Setup config vars
         run: |

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -6,6 +6,7 @@ on:
   push:
     tags:
       - 'v*'
+  pull_request:
 
 name: Create Release
 
@@ -17,10 +18,18 @@ jobs:
       - uses: actions/checkout@v2
       - name: Extract tag name
         id: get_tag
-        run: echo "::set-output name=tag::${GITHUB_REF#refs/tags/}"
+        run: |
+          echo "Running for $GITHUB_EVENT_NAME event"
+          if [[ "$GITHUB_EVENT_NAME" == "push" ]]; then
+            echo "::set-output name=tag::${GITHUB_REF#refs/tags/}"
+          else
+            version=$(grep "set(_version " CMakeLists.txt | head -n1 | sed 's/^.* \([0-9]*\.[0-9]*\.[0-9]*\).*$/\1/')
+            echo "::set-output name=tag::v$version"
+          fi
       - name: Sanity check version
         run: |
           version=${{steps.get_tag.outputs.tag}}
+          echo "Expecting version $version"
           if ! grep -q "set(_version ${version#v})" CMakeLists.txt; then
             echo "Version mismatch."
             echo "Expected '${version#v}', found '$(grep -E 'set\(_version [0-9]' CMakeLists.txt)'"
@@ -39,9 +48,10 @@ jobs:
         run: |
             FOLDER="nowide_${{steps.get_tag.outputs.tag}}"
             mkdir "$FOLDER"
-            cp -r build cmake include src test CMakeLists.txt Config.cmake.in "$FOLDER"
+            cp -r build cmake config include src test CMakeLists.txt Config.cmake.in "$FOLDER"
             tar -czf nowide.tar.gz "$FOLDER"
       - name: Create Release
+        if: github.event_name == 'push'
         id: create_release
         uses: actions/create-release@v1
         env:
@@ -54,6 +64,7 @@ jobs:
           draft: false
           prerelease: false
       - name: Upload standalone version
+        if: github.event_name == 'push'
         uses: actions/upload-release-asset@v1.0.1
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
@@ -63,6 +74,7 @@ jobs:
           asset_name: nowide_standalone_${{steps.get_tag.outputs.tag}}.tar.gz
           asset_content_type: application/tar.gz
       - name: Upload Boost version
+        if: github.event_name == 'push'
         uses: actions/upload-release-asset@v1.0.1
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
@@ -72,6 +84,7 @@ jobs:
           asset_name: nowide_${{steps.get_tag.outputs.tag}}.tar.gz
           asset_content_type: application/tar.gz
       - name: Upload Documentation
+        if: github.event_name == 'push'
         uses: actions/upload-release-asset@v1.0.1
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -14,6 +14,9 @@ jobs:
   release:
     name: Create Release
     runs-on: ubuntu-latest
+    env:
+      DEP_DIR: ${{github.workspace}}/dependencies
+      BOOST_VERSION: 1.56.0
     steps:
       - uses: actions/checkout@v2
       - name: Extract tag name
@@ -50,6 +53,18 @@ jobs:
             mkdir "$FOLDER"
             cp -r build cmake config include src test CMakeLists.txt Config.cmake.in "$FOLDER"
             tar -czf nowide.tar.gz "$FOLDER"
+      - name: Install boost
+        run: sudo apt-get install -y libboost-all-dev
+      - name: Test Boost release tarball
+        run: |
+          tmp_dir=$(mktemp -d -p "$RUNNER_TEMP")
+          cd "$tmp_dir"
+          tar -xf "${{github.workspace}}/nowide.tar.gz"
+          src_dir="$PWD/nowide_${{steps.get_tag.outputs.tag}}"
+          mkdir build && cd build
+          cmake "$src_dir" -DBoost_DEBUG=ON -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=${{runner.workspace}}/../install
+          cmake --build . --config Debug --target install
+          ctest --output-on-failure -C Debug --verbose
       - name: Create Release
         if: github.event_name == 'push'
         id: create_release

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -15,7 +15,7 @@
 
 cmake_minimum_required(VERSION 3.9)
 # Version number starts at 10 to avoid conflicts with Boost version
-set(_version 11.1.2)
+set(_version 11.1.3)
 if(BOOST_SUPERPROJECT_SOURCE_DIR)
   set(_version ${BOOST_SUPERPROJECT_VERSION})
 endif()

--- a/doc/changelog.dox
+++ b/doc/changelog.dox
@@ -10,6 +10,11 @@
 
 \section changelog Changelog
 
+\subsection changelog_11_1_3 Nowide 11.1.3
+
+- Fix missing config file in release
+- Known issues: Read performance for text files is degraded. Binary files and writing is unaffected.
+
 \subsection changelog_11_1_2 Nowide 11.1.2 (Boost 1.76)
 
 - Don't define `__MSVCRT_VERSION__` version to fix compatibility with ucrt

--- a/test/test_fs.cpp
+++ b/test/test_fs.cpp
@@ -1,5 +1,6 @@
 //
 //  Copyright (c) 2015 Artyom Beilis (Tonkikh)
+//  Copyright (c) 2021 Alexander Grund
 //
 //  Distributed under the Boost Software License, Version 1.0. (See
 //  accompanying file LICENSE or copy at
@@ -12,6 +13,9 @@
 #include <boost/nowide/cstdio.hpp>
 #include <boost/nowide/fstream.hpp>
 #include "test.hpp"
+#if defined(_MSC_VER)
+#pragma warning(disable : 4714) // function marked as __forceinline not inlined
+#endif
 #include <boost/filesystem/operations.hpp>
 
 // coverity [root_function]

--- a/test/test_fs.cpp
+++ b/test/test_fs.cpp
@@ -7,6 +7,9 @@
 //  http://www.boost.org/LICENSE_1_0.txt)
 //
 
+#if defined(__GNUC__) && __GNUC__ >= 7
+#pragma GCC diagnostic ignored "-Wattributes"
+#endif
 #include <boost/nowide/filesystem.hpp>
 
 #include <boost/nowide/convert.hpp>

--- a/test/test_traits.cpp
+++ b/test/test_traits.cpp
@@ -30,6 +30,9 @@
 #endif
 
 #ifdef BOOST_NOWIDE_TEST_BFS_PATH
+#if defined(_MSC_VER)
+#pragma warning(disable : 4714) // function marked as __forceinline not inlined
+#endif
 #include <boost/filesystem/path.hpp>
 #endif
 

--- a/test/test_traits.cpp
+++ b/test/test_traits.cpp
@@ -30,6 +30,9 @@
 #endif
 
 #ifdef BOOST_NOWIDE_TEST_BFS_PATH
+#if defined(__GNUC__) && __GNUC__ >= 7
+#pragma GCC diagnostic ignored "-Wattributes"
+#endif
 #if defined(_MSC_VER)
 #pragma warning(disable : 4714) // function marked as __forceinline not inlined
 #endif

